### ISSUE TITLE
Preferred video format

### ISSUE THREAD

**Issue by dimqua** [iv-org/invidious#97]
I think that `Preferred video quality` option should be renamed to `Preferred video resolution`. Then you can add `Preferred video format` option. In some situation when both webm and mp4 formats are available, the automatic choice is unpredictable.

**Comment by omarroth** [comment]
I'm mostly just basing that off of YouTube, which calls it `quality`. I'm not sure if there's a need for a `video format`, I think it would mostly just be confusing to users the difference between `resolution` and `format`. Especially once #34 is implemented (which should be fairly soon), I don't think `preferred format` would serve much purpose.

**Comment by dimqua** [comment]
I agree that suggested implementation is way too complicated than it should (at least for YouTube videos). It can be much simpler:

![](https://i.imgur.com/PIeWob9.png)

**Comment by rezad1393** [comment]
adding a choice for video format (preferably a order like av1-->vp9-->h264) set by user would be great.
when I watch this video https://www.youtube.com/watch?v=42DUA51FXOU on youtube it uses av1 and on invidious it uses h264 and not even vp9, and quality difference is noticeable.

**Comment by rezad1393** [comment]
bumb

**Comment by KweezyCode** [comment]
any updates?

**Comment by unixfox** [comment]
Closing in favor of https://github.com/iv-org/invidious/issues/4847

**Comment by KweezyCode** [comment]
> Closing in favor of #4847

i mean, is is still not possible to choose h264/av1 as option?

**Comment by unixfox** [comment]
> > Closing in favor of #4847
> 
> i mean, is is still not possible to choose h264/av1 as option?

that's a different issue: https://github.com/iv-org/invidious/issues/3012
